File: frontend/src/app/models/product.ts
import { Category } from "./category";
import { Review } from "./review";

export interface Product {
    id: number,
    name: string,
    description: string,
    price: number,
    stock: number,
    average: number,
    images: string[],
    categoryId: number,
    category: Category,
    reviews: Review[],
    total: number // Para el carrito
}

File: backend/Server/Server/Models/Product.cs
﻿using System;
using System.Collections.Generic;

namespace Server.Models;

public class Product
{
    public string Name { get; set; }

    public int Id { get; set; }

    public string Description { get; set; }

    public long Price { get; set; }

    public int Stock { get; set; }

    public double Average { get; set; } 

    public List<string> Images {  get; set; } = new List<string>();

    public int CategoryId { get; set; }

    public Category Category { get; set; }

    public ICollection<Review> Reviews { get; set; } = new List<Review>();
}
